src/Model/LanguageFeatures.re

```diff
@@ -0,0 +1,77 @@
+/*
+ * LanguageFeatures.re
+ *
+ * This module documents & types the protocol for communicating with the VSCode Extension Host.
+ */
+
+open Oni_Core;
```

Comment:
Add REI file

Reply:
Why is this a module? Is there an actual (future) need for it, or is it just because VSCode/LSP organizes it this way? Could it not just be a `Completions` module and have other "language features" as separate modules as well?

Reply:
> Why is this a module? (Edit: Or rather, why is it called LanguageFeatures?) Is there an actual (future) need for it, or is it just because VSCode/LSP organizes it this way? 

Just seemed like a natural grouping for now - I could see splitting it up though (maybe in the future as we have more in here, and can see the boundaries clearly). I was going to add some other 'language features' here:
- DefinitionProvider
- FindAllReferencesProvider
- DocumentHighlightsProvider
- DocumentSymbolProvider
- WorkspaceSymbolProvider
etc

> Could it not just be a `Completions` module and have other "language features" as separate modules as well?
We actually have a `Completions` module right now, that tracks the state of completion _items_. So we're differentiating between the 'provider' (the strategy for obtaining completions), vs the actual completions available to the user. 

It might be though we could split it out into a `CompletionsProvider` module - but I'd prefer to defer this until we have more examples here and a better overview of the constraints for refactoring.

Reply:
Cool, it's definitely not a big deal, but if there' such a thing as "premature organization" I think this might be it :wink: It just seems to me to add indirection for no real purpose, and I guess I tend to start in the other end, creating individual pieces that I might later compose if needed.

Reply:
Cool, thanks for the feedback! I'll track revisiting it here: https://github.com/onivim/oni2/projects/10#card-30449023